lib/internal/modules/esm/load.js

```diff
@@ -69,12 +70,35 @@ async function getSource(url, context) {
   return { __proto__: null, responseURL, source };
 }
 
+function getSourceSync(url, context) {
+  const parsed = new URL(url);
+  const responseURL = url;
+  let source;
+  if (parsed.protocol === 'file:') {
+    source = readFileSync(parsed);
+  } else if (parsed.protocol === 'data:') {
```

Comment:
`URL::protocol` (and all props not `href`) is computed each read, so we can get a bit of a perf improvement by caching its value (which isn't changed):

```suggestion
  const parsed = new URL(url);
  const { protocol } = parsed;
  const responseURL = url;
  let source;
  if (protocol === 'file:') {
    source = readFileSync(parsed);
  } else if (protocol === 'data:') {
```

also, nit: i think this would be more grokkable as a `switch` because we immediately know all cases are related to `protocol`.

Reply:
I copied that from `defaultLoad`, we should probably address both in a follow up PR.